File: Serene/Serene.Script/Imports/FormContexts/Northwind.TerritoryForm.cs

namespace Serene.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Runtime.CompilerServices;

    public partial class TerritoryForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Northwind.Territory";
    
        public TerritoryForm(string idPrefix) : base(idPrefix) {}
    
        public StringEditor TerritoryID { [InlineCode("{this}.w('TerritoryID', Serenity.StringEditor)")] get; private set; }
        public StringEditor TerritoryDescription { [InlineCode("{this}.w('TerritoryDescription', Serenity.StringEditor)")] get; private set; }
        public LookupEditor RegionID { [InlineCode("{this}.w('RegionID', Serenity.LookupEditor)")] get; private set; }
    }
}

File: Serene/Serene.Web/Modules/Common/Imports/ServerTypings/Northwind.TerritoryForm.ts
namespace Serene.Northwind {
    export class TerritoryForm extends Serenity.PrefixedContext {
        static formKey = 'Northwind.Territory';

    }

    export interface TerritoryForm extends Serenity.PrefixedContext {
        TerritoryID(): Serenity.StringEditor;
        TerritoryDescription(): Serenity.StringEditor;
        RegionID(): Serenity.LookupEditor;
    }

    [['TerritoryID', Serenity.StringEditor], ['TerritoryDescription', Serenity.StringEditor], ['RegionID', Serenity.LookupEditor]].forEach(x => TerritoryForm.prototype[<string>x[0]] = function() { return this.w(x[0], x[1]); });
}

